gcloud/_helpers.py

```diff
@@ -188,12 +188,19 @@ def _default_service_project_id():
     :rtype: str or ``NoneType``
     :returns: Project-ID from default configuration file else ``None``
     """
-    full_config_path = os.path.expanduser(DEFAULT_CONFIGURATION_PATH)
+    search_paths = []
+    # Workaround for GAE not supporting pwd which is used by expanduser.
+    try:
+        full_config_path = os.path.expanduser(DEFAULT_CONFIGURATION_PATH)
+        search_paths.append(full_config_path)
```

Comment:
You don't need `full_config_path` except to jam it into `search_paths`, right?  Could just be:

```python
    try:
        search_paths.append(os.path.expanduser(DEFAULT_CONFIGURATION_PATH)
    except ImportError:
        pass
```